src/BitBuffer.cpp

```diff
@@ -204,23 +204,44 @@ uint8_t BitBuffer::read_byte ()
 {
 	uint8_t data = 0;
 
-	if (encoding == Encoding::FM)
+	switch (encoding)
 	{
+	case Encoding::FM:
 		for (auto i = 0; i < 8; ++i)
 		{
 			read1();
 			read1();
 			data = (data << 1) | read1();
 			read1();
 		}
-	}
-	else
-	{
+		break;
+
+	case Encoding::MFM:
 		for (auto i = 0; i < 8; ++i)
 		{
 			read1();
 			data = (data << 1) | read1();
 		}
+		break;
+
+	case Encoding::Apple:
+		for (auto i = 0; i < 8; ++i)
+		{
+			data = (data << 1) | read1();
+		}
+		// Disk ][ keeps reading until bit 7 is 1
+		for (;(data & 0x80) == 0;)
```

Comment:
Could be worth including a wrapped() check here to prevent it getting stuck if there are no '1' bits?